src/kernels/PrecursorEigenSource.C

```diff
@@ -0,0 +1,87 @@
+#include "PrecursorEigenSource.h"
+
+registerMooseObject("MoltresApp", PrecursorEigenSource);
+
+InputParameters
+PrecursorEigenSource::validParams()
+{
+  InputParameters params = EigenKernel::validParams();
+  params += ScalarTransportBase::validParams();
+  params.addRequiredParam<unsigned int>("num_groups", "The total number of energy groups");
+  params.addRequiredCoupledVar("group_fluxes", "All the variables that hold the group fluxes. "
+                                               "These MUST be listed by decreasing "
+                                               "energy/increasing group number.");
+  params.addParam<unsigned int>("precursor_group_number",
+                                "What precursor group this kernel is acting on.");
+  params.addCoupledVar(
+      "temperature", 800, "The temperature used to interpolate material properties.");
+  params.addParam<Real>("prec_scale", 1, "Thefactor by which the neutron fluxes are scaled.");
+  return params;
+}
+
+PrecursorEigenSource::PrecursorEigenSource(const InputParameters & parameters)
```

Comment:
Maybe add a docstring (nto sure if it should go here or in the header file)? It's hard for me to tell what this function does. Also, it's been a while since I've done anything in C, but `& parameters` is a pointer, right?

Reply:
The code structure in this file follows the standard MOOSE format for Kernels. This function initializes requisite member variables in `PrecursorEigenSource` for subsequent calculations in `computeQpResidual`, `computeQpJacobian`, etc. Yea `& parameters` is a pointer.
I'll add docstrings through the `validParams` function and in the header file as is typical for MOOSE kernels.